src/ontodia/editor/serializedDiagram.ts

```diff
@@ -0,0 +1,181 @@
+import { pick } from 'lodash';
+
+import { ElementIri, LinkTypeIri } from '../data/model';
+
+import { Element as DiagramElement, Link as DiagramLink } from '../diagram/elements';
+import { Vector, Size } from '../diagram/geometry';
+
+export interface SerializedDiagram {
+    '@context': any;
+    '@type': 'Diagram';
+    layoutData: LayoutData;
+    linkTypeOptions: LinkTypeOptions[];
+}
+
+export interface LinkTypeOptions {
+    '@type': 'LinkTypeOptions';
+    property: LinkTypeIri;
```

Comment:
I'm doing a conversion between old and new structure in convertToLatest(). I'll check if I do it properly for the links. 'property' is better then 'id' because it will be consistent with resulting RDF. Id is the object's IRI and this LinkTypeOption's id references another entity.